endaq/calc/utils.py

```diff
@@ -206,3 +211,49 @@ def _rolling_slice_definitions(
     num = int(slice_width / spacing / 2)
 
     return indexes, slice_width, num, length
+
+
+def convert_units(
+        src: str,
+        dst: str,
+        df: pd.DataFrame = None,
+        print_conversion: bool = False,
+) -> pd.DataFrame:
+    """
+    Using the `Pint library <https://github.com/hgrecco/pint/blob/master/pint/default_en.txt>`_ apply a unit
+    conversion to a provided unit-unaware dataframe.
+    :param src: a text string defining the base units to convert from like `"in"` for inches
+    :param dst: a text string defining the destination units to convert to like `"mm"` for millimeters
+    :param df: the input dataframe, if none is provided the unit conversion is only applied from `src` to `dst`
+    :param print_conversion: if `True` the conversion applied is printed to the console, default is `False`
+    :return: a dataframe with the values scaled according to the unit conversion, if no dataframe is provided then a
+        scaler value is returned
+    Some examples are provided below:
+    .. code-block:: python
```

Comment:
For proper Sphinx formatting, I think you need blank lines on either side of the "Some examples" line. Probably before the first parameter (`:param src:`) as well.